idseq_dag/util/command_patterns.py

```diff
@@ -82,6 +87,9 @@ def as_dict(self) -> Dict[str, Any]:
         result.update(super().as_dict())
         return result
 
+    def as_test_str(self):
```

Comment:
Not great adding functions like this just for testing. You should be able to mock subprocess to capture the call.

Reply:
I see the weakness in testing something that is not actually executed in production. 

Instead, what do you think of testing `_command_args` method?

Reply:
You could a `expect(command).to receive(:execute).with(...)`. You can create the expected SingleCommand. It is not you job to test the command/command_patterns modules.
Would that work?

Reply:
I agree with Tiago. Also, this method of testing is not good, since it will assume these two things are the same:
`SingleCommand(cmd="grep", args=["a b", "c.txt"])`
and
`SingleCommand(cmd="grep", args=["a", "b c.txt"])`

You could check the results of `as_dict` eventually, or `_command_args` (although this is a protected method). You can also direct test `cmd` and `args` properties since they are public.

And finally, we can eventually write a custom matcher, that could be reused elsewhere, but this is slightly harder than just checking these properties.

Reply:
Thanks for the input. I will remove the `as_test_str` methods and test the `cmd` and `args`.